src/event_pusher/mod_event_pusher_push.erl

```diff
@@ -323,3 +248,53 @@ parse_form(Form) ->
             invalid_form
     end.
 
+-spec maybe_enable_node(jid:jid(), jid:jid(), pubsub_node(), form(), jlib:iq()) -> jlib:iq().
+maybe_enable_node(#jid{lserver = Host} = From, BarePubSubJID, Node, FormFields, IQ) ->
+    AllKnownDomains = ejabberd_router:dirty_get_all_domains() ++ virtual_pubsub_hosts(Host),
+    case lists:member(BarePubSubJID#jid.lserver, AllKnownDomains) of
+        true ->
+            ok = mod_event_pusher_push_backend:enable(jid:to_bare(From), BarePubSubJID, Node, FormFields),
+            ejabberd_sm:store_info(From, {push_notifications, {Node, FormFields}}),
+            IQ#iq{type = result, sub_el = []};
+        false ->
+            NewSubEl = [IQ#iq.sub_el, mongoose_xmpp_errors:remote_server_not_found()],
+            IQ#iq{type = error, sub_el = NewSubEl}
+    end.
+
+
+-spec maybe_remove_push_node_from_sessions_info(jid:jid(), pubsub_node() | undefined) ->
+    ok.
+maybe_remove_push_node_from_sessions_info(From, undefined) ->
+    %% The node is undefined which means that a user wants to disable all push nodes
+    AllResources = ejabberd_sm:get_user_present_resources(From),
+    [begin
+         JID = jid:replace_resource(From, Resource),
+         ejabberd_sm:remove_info(JID, push_notifications)
+     end || {_, Resource} <- AllResources],
+    ok;
+maybe_remove_push_node_from_sessions_info(From, Node) ->
+    AllSessions = ejabberd_sm:get_raw_sessions(From),
+    find_and_remove_push_node(From, AllSessions, Node).
+
+-spec find_and_remove_push_node(jid:jid(), [ejabberd_sm:session()], pubsub_node()) -> ok.
+find_and_remove_push_node(_From, [], _Node) ->
+    ok;
+find_and_remove_push_node(From, [RawSession | Rest], Node) ->
+    case my_push_node(RawSession, Node) of
+        true ->
+            LResource  = mongoose_session:get_resource(RawSession),
+            JID = jid:replace_resource(From, LResource),
+            ejabberd_sm:remove_info(JID, push_notifications);
+        false ->
+            find_and_remove_push_node(From, Rest, Node)
+    end.
+
+-spec my_push_node(ejabberd_sm:session(), pubsub_node()) -> boolean().
+my_push_node(RawSession, Node) ->
+    SInfo = mongoose_session:get_info(RawSession),
+    case lists:keyfind(push_notifications, 1, SInfo) of
+        {push_notifications, {Node, _}} ->
+            true;
+        _ ->
+            false
+    end.
```

Comment:
No new line.

Reply:
if you don't mind I will add it in the next PR, don't want to hold other